### ISSUE TITLE
EXC_BAD_ACCESS crash inside AppleIntelHD5000GraphicsGLDriver when starting 3.0a8+

### ISSUE THREAD

**Issue by radames** [processing/processing#3359]
I'm using the Mac OSX 10.10.3 and both last alpha versions Processing 3.0a8 and 3.0a9 are crashing on  startup, although the version 3.0a5 works perfectly.

Both with same report, 

```
...
Crashed Thread:        0  AppKit Thread  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGABRT)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000303030353150

VM Regions Near 0x303030353150:
    __LINKEDIT             000012340057b000-00001234006e7000 [ 1456K] r--/rwx SM=COW  /System/Library/Extensions/AppleIntelHD5000GraphicsGLDriver.bundle/Contents/MacOS/AppleIntelHD5000GraphicsGLDriver
--> 
    MALLOC_NANO            0000600000000000-0000600000200000 [ 2048K] rw-/rwx SM=COW  
...
```

**Comment by benfry** [comment]
Can you try running it from the "guest" account on your machine and see if it works?

What type of Mac is this on?

**Comment by prisonerjohn** [comment]
@benfry Do you think this is the same issue I'm having?

My stack trace is a little different:
```
Crashed Thread:        0  AppKit Thread  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGABRT)
Exception Codes:       EXC_I386_GPFLT

Application Specific Information:
abort() called
```

**Comment by benfry** [comment]
It's a suspicion, but doesn't seem like it's crashing in the same spot, so I wanted to see if the same thing was happening from another account. 

In both cases, I'm also wondering if it's a JDK bug—we've been moving to the latest JDK 8 with each release, so it may have more to do with that than anything we've actually changed.

On yours, try changing processing/build/build.xml to use:
```
<property name="jdk.version" value="8" />
<property name="jdk.update" value="25" />
<property name="jdk.build" value="17" />
```
Right now it's at 45 for the jdk.update value, and something else for jdk.build. Try changing it, then `ant clean` and `ant run` and see if it changes anything.

**Comment by benfry** [comment]
Also for @prisonerjohn, where's your original bug report? I can't seem to find it. Does your laptop also have Intel HD 5000 graphics?

**Comment by prisonerjohn** [comment]
#3285 is my original report, but it looks like it was a different issue. 

My laptop has an Intel HD Graphics 4000, but it's never used. I'm always on the NVIDIA GeForce GT 650M.

**Comment by prisonerjohn** [comment]
Tried changing the JDK version values but still crashing:
```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007fff8cf18152, pid=6476, tid=1299
#
# JRE version: Java(TM) SE Runtime Environment (8.0_25-b17) (build 1.8.0_25-b17)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.25-b02 mixed mode bsd-amd64 compressed oops)
# Problematic frame:
# C  [libsystem_c.dylib+0x1152]  strlen+0x12
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /Users/elias/hs_err_pid6476.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.sun.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
```
Crash log is [here](https://dl.dropboxusercontent.com/u/477840/hs_err_pid6476.log).

**Comment by radames** [comment]
Hi @benfry , I've tried to open with the Guest account and had the same problem. I'm on osx 10.10.3 using Java 7 Update 75.

I'll try to compile it later

**Comment by benfry** [comment]
Alright, @prisonerjohn I've re-filed yours here: https://github.com/processing/processing/issues/3360. The error message shown in that other one has nothing to do with the hard crash, so I don't want to confuse things further. Let's continue the discussion over there.

This appears to be a separate issue.

**Comment by benfry** [comment]
@radames Don't worry about building, it's not necessary. But out of curiosity, can you try to replace the `Processing.app/Contents/MacOS/Processing` file in 3.0a9 with the one from 3.0a5 and try running that?

**Comment by radames** [comment]
@benfry yes I did the replacement and the 3.0a9 seems working fine, I've tried some examples using shaders and seems compiling and working.

**Comment by radames** [comment]
even the debug mode works fine!

**Comment by benfry** [comment]
Here are two versions of that Processing file. Can you try each and see if either or both work properly?
https://www.dropbox.com/s/t0eggec8o4s6v9s/alternates.zip?dl=0

Unzip that file, and use Processing-main to replace the Processing file at the location listed earlier. Try running, see if it works. Then do the same with the other. 

Thanks!

**Comment by radames** [comment]
@benfry I did try for both files neither of then worked, had the same crash message.
I assured the change the execute permissions.

**Comment by benfry** [comment]
Ouch, ok. That's the worst of the scenarios that could have happened. One version is the exact same code as 3.0a5 (just recompiled on 10.10.3). In the meantime, we'll roll back to the 3.0a5 version, at any rate.

**Comment by benfry** [comment]
Thanks for checking into it. On further digging, it looks like there were some other edits (and a likely culprit) since 3.0a5. Can you try [this version](https://www.dropbox.com/s/u3bvqnlrq0eeka9/alternate2.zip?dl=0)?

**Comment by radames** [comment]
Same, problem with your last version.

```
Crashed Thread:        0  AppKit Thread  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGABRT)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000303030353150

VM Regions Near 0x303030353150:
    __LINKEDIT             000012340057b000-00001234006e7000 [ 1456K] r--/rwx SM=COW  /System/Library/Extensions/AppleIntelHD5000GraphicsGLDriver.bundle/Contents/MacOS/AppleIntelHD5000GraphicsGLDriver
--> 
    MALLOC_NANO            0000600000000000-0000600000200000 [ 2048K] rw-/rwx SM=COW 
```

Anyway it still working with the one file taken from Processing 3.0a5

**Comment by benfry** [comment]
Nuts, ok. Thought we had it. Will continue looking...

**Comment by radames** [comment]
Let me know how can I help. One question, I'll be starting teaching some class this week and I'd like to use this last alpha version, do you think the trick of moving the file from the 3.0a5 is a good idea in order to try the new 3.0a9?

**Comment by benfry** [comment]
That's definitely a fix—I was hoping to have a new release out this past weekend but that was held up in part by this issue, since it's obviously quite serious (though only causes trouble for a small percentage of people). I'm still not sure why this is showing up on some machines and not others, but am hoping to find a better fix in the next 24-48 hours.

**Comment by radames** [comment]
that's great! thanks!

**Comment by benfry** [comment]
There's a workaround in 3.0a10 for this (rolling back to an older, buggier native app), but we still need to figure out what's going on here.

**Comment by radames** [comment]
yes, I'm already using it, thanks! so far it's been ok! some of my students are using the same version for windows and they're facing some problems, but still need to help them to report it here!

**Comment by benfry** [comment]
Fixed in that earlier alpha, but new issues introduced; see #3790.
